source/xcb.c

```diff
@@ -1375,6 +1404,15 @@ static gboolean main_loop_x11_event_handler(xcb_generic_event_t *ev,
   if (xcb->sndisplay != NULL) {
     sn_xcb_display_process_event(xcb->sndisplay, ev);
   }
+
+  if (xcb->im && xcb_xim_filter_event(xcb->im, ev) == 0) {
```

Comment:
So now, if `xcb_xim_filter_even()` returns `true` (which is, IIUC, “I filtered something, please ignore the event”) you still continue?  
I think it should be
```c
if (xcb->im && xcb_xim_filter_event(xcb->im, ev) ) {
    return G_SOURCE_CONTINUE;
}

/* In the big switch */
    case XCB_KEY_PRESS:
        if ( xcb->ic ) {
            xcb_xim_forward_event(xcb->im, xcb->ic, (xcb_key_press_event_t *)event);
        } else {
            rofi_key_press_event_handler(event, state);
        }
 /* same for KEY_RELEASE */
```